### DISCUSSION TITLE
What is the equivalent of continue or break operator in cuelang

### DISCUSSION THREAD

**Discussion by xi3161-sregoti** [cue-lang/cue#1754] [Q&A]
Is there any equivalent of `continue` or `break` operator of golang in cue. I have dublicate conditions that i want to avoid

Below is my cue code
```
import (
    "example.com/pkg/schemas:transform"
    "example.com/pkg/schemas:track"
    "example.com/pkg/schemas:input"
)


outpus: [
    for i, ti in transform.transformedInputs
        for j, tm in track.modules
            if ti.name == tm.name {
                {
                    a: 1
                }
            }
]
```

Which gives the below output
```
{
    "outputs": [
        {
            "a": 1
        },
        {
            "a": 1
        },
        {
            "a": 1
        }
    ]
}
```

**Comment by mvdan** [comment]
I don't believe there are continue nor break operators, but it should be noted that your example can be simplified quite a bit. You don't need the last `{}` struct nesting, and if you `cue fmt`, you will notice that two more levels of indentation are gone as well:

```
$ cat f.cue
_foo: ["foo1", "foo2", "foo3"]
_bar: ["bar1", "foo2", "bar3", "foo4"]

outputs: [
	for i, ti in _foo
	for j, tm in _bar
	if ti == tm {
		a: 1
	}
]
$ cue export f.cue
{
    "outputs": [
        {
            "a": 1
        }
    ]
}
```

As a self-contained example of CUE code, it seems perfectly fine and I don't directly see the need to write the logic in another way.

Could you provide an example that shows the problem you're facing? You mention duplicate conditions, but your original example doesn't have any.
